### PR TITLE
Code cleaning: clang-format, includes, etc.

### PR DESCRIPTION
**Pull request by jhanca-robotecai**
Code cleaning before the release:
- soft includes into framework/o3de/other groups
- move uuid identifiers of all components into a separate header files
- fix license
- clang-format

### CODE REVIEW

File: Gems/GenAIAmazonBedrockVendorBundle/Code/Source/Providers/Claude/ClaudeAmazonBedrockProvider.h [hunk 1]
```diff
@@ -7,11 +7,13 @@
  */
 
 #pragma once
+#include <GenAIAmazonBedrockVendorBundle/GenAIAmazonBedrockVendorBundleTypeIds.h>
```
Comment (arturkamieniecki) [inline Gems/GenAIAmazonBedrockVendorBundle/Code/Source/Providers/Claude/ClaudeAmazonBedrockProvider.h:10]:
```suggestion

#include <GenAIAmazonBedrockVendorBundle/GenAIAmazonBedrockVendorBundleTypeIds.h>
```

Nit: newline after #pragma once

File: Gems/GenAIFramework/Code/Include/GenAIFramework/GenAIFrameworkTypeIds.h [hunk 2]
```diff
@@ -13,9 +13,19 @@ namespace GenAIFramework
...
 
     inline constexpr const char* GenAIAsyncRequestSystemComponentTypeId = "{D2D195A2-C720-4961-9AEF-C00FCB0AA70D}";
-    inline constexpr const char* GenAIAsyncRequestEditorSystemComponentTypeId = "{018e7a45-43ae-7e69-8bd9-0751b9170bf5}";
+    inline constexpr const char* GenAIAsyncRequestEditorSystemComponentTypeId = "{018E7A45-43AE-7E69-8BD9-0751B9170BF5}";
+
+    // Communication TypeIds
+    inline constexpr const char* SystemRegistrationContextTypeId = "{E162937C-0177-4CED-87B3-3B037A44C394}";
+    inline constexpr const char* AsyncRequestsTypeId = "{018E7A12-5DB2-7DAA-A764-78D7703C8EFA}";
+    inline constexpr const char* AIServiceProviderRequestsTypeId = "{0BCAC4E2-09A9-4702-98DD-EF37006664A7}";
+    inline constexpr const char* AIModelRequestTypeId = "{8C140E10-E3A2-478B-9886-C7C24EB16816}";
```
Comment (arturkamieniecki) [inline Gems/GenAIFramework/Code/Include/GenAIFramework/GenAIFrameworkTypeIds.h:25]:
```suggestion
    inline constexpr const char* AIModelRequestsTypeId = "{8C140E10-E3A2-478B-9886-C7C24EB16816}";
```
Nit: changing the typeid name to be more consistent with AIModelRequests change

File: Gems/GenAIFramework/Code/Include/GenAIFramework/Communication/AIModelRequestBus.h [hunk 3]
```diff
@@ -34,10 +36,10 @@ namespace GenAIFramework
     using ModelAPIExtractedResponse = AZ::Outcome<AIMessage, AZStd::string>; //!< The type of extracted response from the model
     using AIHistory = AIMessages; //!< The type of AI history
 
-    class AIModelRequest : public AZ::ComponentBus
+    class AIModelRequests : public AZ::ComponentBus
     {
     public:
-        AZ_RTTI(AIModelRequest, "{8c140e10-e3a2-478b-9886-c7c24eb16816}", AZ::ComponentBus);
+        AZ_RTTI(AIModelRequests, AIModelRequestTypeId, AZ::ComponentBus);
```
Comment (arturkamieniecki) [inline Gems/GenAIFramework/Code/Include/GenAIFramework/Communication/AIModelRequestBus.h:42]:
```suggestion
        AZ_RTTI(AIModelRequests, AIModelRequestsTypeId, AZ::ComponentBus);
```
Nit: changing the typeid name to be more consistent with AIModelRequests change

File: Gems/GenAIFramework/Code/Source/Clients/GenAIAsyncRequestSystemComponent.h [hunk 4]
```diff
@@ -8,14 +8,15 @@
 
 #pragma once
 
+#include <GenAIFramework/Communication/AsyncRequestBus.h>
+
 #include "AzCore/Component/EntityId.h"
```
Comment (arturkamieniecki) [inline Gems/GenAIFramework/Code/Source/Clients/GenAIAsyncRequestSystemComponent.h:13]:
```suggestion
#include <AzCore/Component/EntityId.h>
```

File: Gems/GenAIVendorBundle/Code/Include/GenAIVendorBundle/GenAIVendorBundleTypeIds.h [hunk 5]
```diff
@@ -19,4 +19,19 @@ namespace GenAIVendorBundle
...
     // Interface TypeIds
     inline constexpr const char* GenAIVendorBundleRequestsTypeId = "{F1D5E6A3-0D13-46F9-8B9B-92A85A145625}";
+
+    // Models TypeIds
+    inline constexpr const char* ClaudeModelConfigurationTypeId = "{5A478AB3-8AC6-4C23-8768-03C2791D6B11}";
+    inline constexpr const char* ClaudeModelMessagesAPITypeId = "{9FFED8C5-A872-4ADF-BC80-74BA1FE92E95}";
+    inline constexpr const char* ClaudeModelTextCompletionsTypeId = "{0310426D-3035-4240-B9E5-D1556AF98B47}";
+    inline constexpr const char* OllamaChatModelConfigurationTypeId = "{CE2D1912-94EE-4BCD-9015-E118900F4051}";
+    inline constexpr const char* OllamaChatModelTypeId = "{E82C0C54-0658-419A-8170-033EF06AEA96}";
+    inline constexpr const char* OllamaModelConfigurationTypeId = "{C5FA34FF-2BBB-4DF4-9AA2-C27320B57F93}";
```
Comment (arturkamieniecki) [inline Gems/GenAIVendorBundle/Code/Include/GenAIVendorBundle/GenAIVendorBundleTypeIds.h:29]:
```suggestion
    inline constexpr const char* OllamaGenerateModelConfigurationTypeId = "{C5FA34FF-2BBB-4DF4-9AA2-C27320B57F93}";
```

This PR is a good place to add the Generate name to the OllamaModel components. The generate name is used in the ollama documentation as a name in the URL for ollama serve: https://github.com/ollama/ollama?tab=readme-ov-file#rest-api.

File: Gems/GenAIVendorBundle/Code/Include/GenAIVendorBundle/GenAIVendorBundleTypeIds.h [hunk 6]
```diff
@@ -19,4 +19,19 @@ namespace GenAIVendorBundle
...
+    inline constexpr const char* OllamaModelTypeId = "{3782988D-058F-4943-9862-874EBC90A240}";
```
Comment (arturkamieniecki) [inline Gems/GenAIVendorBundle/Code/Include/GenAIVendorBundle/GenAIVendorBundleTypeIds.h:30]:
```suggestion
    inline constexpr const char* OllamaGenerateModelTypeId = "{3782988D-058F-4943-9862-874EBC90A240}";
```
This PR is a good place to add the Generate name to the OllamaModel components. The generate name is used in the ollama documentation as a name in the URL for ollama serve: https://github.com/ollama/ollama?tab=readme-ov-file#rest-api.

**Review by arturkamieniecki — APPROVED** [review]
Some nits. I would suggest changing the OllamaModel name to OllamaGenerateModel. As per the REST API used in Ollama github: https://github.com/ollama/ollama?tab=readme-ov-file#rest-api.

File: Gems/GenAIVendorBundle/Code/Include/GenAIVendorBundle/GenAIVendorBundleTypeIds.h (hunk 5 above)
Comment (arturkamieniecki) [reply Gems/GenAIVendorBundle/Code/Include/GenAIVendorBundle/GenAIVendorBundleTypeIds.h:29]: Or creating a seperate PR.

File: Gems/GenAIVendorBundle/Code/Include/GenAIVendorBundle/GenAIVendorBundleTypeIds.h (hunk 6 above)
Comment (arturkamieniecki) [reply Gems/GenAIVendorBundle/Code/Include/GenAIVendorBundle/GenAIVendorBundleTypeIds.h:30]: Or creating a seperate PR.

File: Gems/GenAIVendorBundle/Code/Include/GenAIVendorBundle/GenAIVendorBundleTypeIds.h (hunk 5 above)
Comment (jhanca-robotecai) [reply Gems/GenAIVendorBundle/Code/Include/GenAIVendorBundle/GenAIVendorBundleTypeIds.h:29]: This should be solved in a separate PR. I created an issue out of your comment.

**Pull request merged**